addons/account/models/account.py

```diff
@@ -634,6 +636,12 @@ def write(self, vals):
                     'refund_sequence_number_next': vals.get('refund_sequence_number_next', journal.refund_sequence_number_next),
                 }
                 journal.refund_sequence_id = self.sudo()._create_sequence(journal_vals, refund=True).id
+        # Changing the 'post_at_bank_rec' option will post the draft payment moves and change the related invoices' state.
+        if 'post_at_bank_rec' in vals and not vals['post_at_bank_rec']:
+            draft_moves = self.env['account.move'].search([('journal_id', '=', self.id), ('state', '=', 'draft')])
+            pending_payments = draft_moves.mapped('line_ids.payment_id')
+            pending_payments.mapped('move_line_ids.move_id').post()
+            pending_payments.mapped('reconciled_invoice_ids').filtered(lambda x: x.state == 'in_payment').write({'state': 'paid'})
```

Comment:
it's plain wrong to mark invoices in 'paid' status: it depends on the amount_residual to keep them in 'open' status or not

Reply:
'in_payment' state is only given when invoice has 'reconciled' set to True (it just replaces the 'paid' state) ; so it's not wrong here ; but we can do otherwise if you want :)